build.gradle.kts

```diff
@@ -0,0 +1,240 @@
+                    developer {
+                        id.set("sonarsource-team")
+                        name.set("SonarSource Team")
+                    }
+                }
+            }
+            artifact(bomArtifact)
+        }
+    }
+}
+
+artifactory {
+    clientConfig.isIncludeEnvVars = true
+    clientConfig.envVarsExcludePatterns =
+        "*password*,*PASSWORD*,*secret*,*MAVEN_CMD_LINE_ARGS*,sun.java.command,*token*,*TOKEN*,*LOGIN*,*login*,*KEY*,*signing*"
+    setContextUrl(System.getenv("ARTIFACTORY_URL"))
+    publish {
+        repository {
+            setRepoKey(System.getenv("ARTIFACTORY_DEPLOY_REPO"))
+            setUsername(System.getenv("ARTIFACTORY_DEPLOY_USERNAME"))
+            setPassword(System.getenv("ARTIFACTORY_DEPLOY_PASSWORD"))
+        }
+        defaults {
+            setProperty(
+                "properties",
+                mapOf(
+                    "build.name" to "sonar-scanner-gradle",
+                    "build.number" to System.getenv("BUILD_NUMBER"),
+                    "pr.branch.target" to System.getenv("PULL_REQUEST_BRANCH_TARGET"),
+                    "pr.number" to System.getenv("PULL_REQUEST_NUMBER"),
+                    "vcs.branch" to System.getenv("GIT_BRANCH"),
+                    "vcs.revision" to System.getenv("GIT_COMMIT"),
+                    "version" to project.version as String,
+                )
+            )
+            publications("pluginMaven")
+            setPublishPom(true)
+            setPublishIvy(false)
+        }
+    }
+    clientConfig.info.buildName = "sonar-scanner-gradle"
+    clientConfig.info.buildNumber = System.getenv("BUILD_NUMBER")
+    // The name of this variable is important because it's used by the delivery process when extracting version from Artifactory build info.
+    clientConfig.info.addEnvironmentProperty("PROJECT_VERSION", version.toString())
+}
+
+tasks.processResources {
+    filesMatching("**/version.txt") {
+        expand(mapOf("version" to project.version))
+    }
+}
+
+signing {
+    val signingKeyId: String? by project
+    val signingKey: String? by project
+    val signingPassword: String? by project
+    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
+    setRequired {
+        val branch = System.getenv()["CIRRUS_BRANCH"] ?: ""
+        (branch == "master" || branch.matches("branch-[\\d.]+".toRegex())) &&
+            gradle.taskGraph.hasTask(":artifactoryPublish")
+    }
+    sign(publishing.publications)
+}
+
+tasks.withType<Sign> {
+    onlyIf {
+        val branch = System.getenv()["CIRRUS_BRANCH"] ?: ""
+        val artifactorySkip: Boolean = tasks.artifactoryPublish.get().skip
+        !artifactorySkip && (branch == "master" || branch.matches("branch-[\\d.]+".toRegex())) &&
+            gradle.taskGraph.hasTask(":artifactoryPublish")
+    }
+}
+
+tasks.withType<Test>().configureEach {
+    doLast {
+        Thread.sleep(2000) // https://github.com/gradle/gradle/issues/16603
+    }
+}
```

Comment:
Minor, but there are a few files, that don't have an empty line at the end, would be nice to add it.